### ISSUE TITLE
Cleanup npm dependencies

### ISSUE THREAD

**Issue by Kagami** [FormidableLabs/radium#292]
Hi. I think it's better to move `babel` and `rimraf` to the `devDependencies` and `babel-core` is superfluous (`babel` package is superset of `babel-core`).

**Comment by ryan-roemer** [comment]
`postinstall` needs to run `lib` which uses `rimraf` and `babel`, so if we don't want to put `lib/` under source, we'll need both.

Also, I just `npm install`'ed only `babel-core`. I don't see anything in `node_modules/.bin`. I _think_ they are needed as separate packages (?)

**Comment by Kagami** [comment]
>Also, I just npm install'ed only babel-core. I don't see anything in node_modules/.bin

It's in `node_modules/radium/node_modules/.bin`

>postinstall needs to run lib which uses rimraf and babel, so if we don't want to put lib/ under source, we'll need both

I see. Could you please explain the point of `postinstall`? You have `prepublish` script and so every npm release should contain the `lib` directory. And npm [doesn't recomment to use](https://docs.npmjs.com/misc/scripts#best-practices) `install` script.

It's not that important, but `babel` is rather big package and keeping it in `dependencies` for no good reason is rather annoying, especially because `radium` itself is lightweight.

**Comment by ryan-roemer** [comment]
@Kagami -- I'll leave the substantive discussion to @alexlande and @ianobermiller of how to build / publish the library. But for:

>  It's in node_modules/radium/node_modules/.bin

I did a clean `npm install --save babel-core` in an empty project and the `babel` binary  did _not_ appear in `node_modules/.bin`. This comports with the what appears to be a "multi-publish" scheme for babel itself:

* [`babel-cli/package.json`](https://github.com/babel/babel/blob/master/packages/babel-cli/package.json -- This is `babel` in NPM and _has_ a `bin` component.
* [`babel/package.json`](https://github.com/babel/babel/blob/master/packages/babel/package.json) - This is `babel-core` in NPM and does not have a `bin` component.

**Comment by Kagami** [comment]
>This comports with the what appears to be a "multi-publish" scheme for babel itself

I'm not sure that I understand.

https://babeljs.io/docs/usage/api/
`babel` is generally = `babel-core` + CLI utility + few other deps. You don't need `babel-core` if you have `babel` (though npm but default deduplicate the `radium -> [babel -> babel-core, babel-core]` dep tree into `radium -> [babel, babel-core]`).

**Comment by ryan-roemer** [comment]
@Kagami -- Sorry, read that the wrong way. I thought you meant get rid of `babel` for `dependencies` (not `babel-core`). Yeah, you may be able to get rid of `babel-core` in favor of `babel` as it _is_ a `dependency` in `babel`.

**Comment by ianobermiller** [comment]
You're right that `babel-core` can be removed. We will keep `babel` and `rimraf` in dependencies for now, though, unless there is a better solution. The reason they exist, and why we have the postinstall hook, is that if you `npm install` radium directly from git, the `lib` directory needs to be built before it can be used.

**Comment by Kagami** [comment]
Thanks for explanation.

>We will keep babel and rimraf in dependencies for now, though, unless there is a better solution

Probably not better, but other solutions for this:

1. Compile ES6 to `lib/` on each commit and push it to the git repo.
2. Just do NPM releases often.

I think a lot of node developers nowadays prefer to do NPM release for every small feature they comitted/bug they fixed (especially since `npm version patch && npm publish` is so simple command) so you will almost never need to install directly from git.

**Comment by scottcorgan** [comment]
FYI, I'm getting "babel-core" is not defined when install master through NPM. I know you're not supposed to do that, but I needed a fix that isn't published to NPM yet.

**Comment by ianobermiller** [comment]
@scottcorgan what command were you running? I just tried the following, and it worked fine:

`npm install https://github.com/FormidableLabs/radium.git`

**Comment by scottcorgan** [comment]
That did work fine for me as well.

It's when I install all my dependencies again after removing my
node_modules folder. I'm pretty sure it has to do with the "postinstall"
script.

**Comment by ianobermiller** [comment]
I still can't repro:

```sh
cd test
npm init -f
npm install --save https://github.com/FormidableLabs/radium.git
rm -rf node_modules
npm install
```

**Comment by scottcorgan** [comment]
Looks like you published 0.13.5, so it won't be an issue anymore. Appreciate it!

**Comment by ianobermiller** [comment]
Well, installing from got should just work, so if you still have a repro
I'd like to fix it.
